pulse2percept/api.py

```diff
@@ -0,0 +1,423 @@
+            time step (as defined by `stim` and `tm`).
+            Default: None.
+        tol : float, optional
+            Ignore pixels whose effective current is smaller than a fraction
+            `tol` of the max value.
+            Default: 0.05.
+
+        Returns
+        -------
+        A utils.TimeSeries object whose data container comprises the predicted
+        brightness over time at each retinal location (x, y), with the last
+        dimension of the container representing time (t).
+
+        Examples
+        --------
+        Stimulate a single-electrode array:
+
+        >>> import pulse2percept as p2p
+        >>> implant = p2p.implants.ElectrodeArray('subretinal', 0, 0, 0, 0)
+        >>> stim = p2p.stimuli.Psycho2Pulsetrain(tsample=5e-6, freq=50, amp=20)
+        >>> sim = p2p.Simulation("Single-electrode example", implant)
+        >>> percept = sim.pulse2percept(stim)  # doctest: +SKIP
+
+        Stimulate a single electrode ('C3') of an Argus I array centered on the
+        fovea:
+
+        >>> import pulse2percept as p2p
+        >>> implant = p2p.implants.ArgusI()
+        >>> stim = {'C3': p2p.stimuli.Psycho2Pulsetrain(tsample=5e-6, freq=50,
+        ...                                              amp=20)}
+        >>> sim = p2p.Simulation("ArgusI example", implant)
+        >>> resp = sim.pulse2percept(stim, implant)  # doctest: +SKIP
+        """
+        self._init()
+
+        # Parse `stim` (either single pulse train or a list/dict of pulse
+        # trains), and generate a list of pulse trains, one for each electrode
+        pt_list = p2p.stimuli.parse_pulse_trains(stim, self.implant)
+
+        self.gcl.tsample = pt_list[0].tsample
+
+        # Perform any necessary calculations per electrode
+        pt_list = p2p.utils.parfor(self.gcl.calc_per_electrode,
+                                   pt_list, engine=self.engine,
+                                   n_jobs=self.num_jobs)
+
+        # Which layer to simulate is given by implant type.
+        # For now, both implant types process the same two layers. In the
+        # future, these layers might differ. Order doesn't matter.
+        if self.implant.etype == 'epiretinal':
+            dolayers = ['NFL', 'INL']  # nerve fiber layer
+        elif self.implant.etype == 'subretinal':
+            dolayers = ['NFL', 'INL']  # inner nuclear layer
+        else:
+            e_s = "Supported electrode types are 'epiretinal', 'subretinal'"
+            raise ValueError(e_s)
+
+        # Derive the effective current spread
+        if self.ofl_streaks:
+            ecs, _ = self.ofl.electrode_ecs(self.implant)
+        else:
+            _, ecs = self.ofl.electrode_ecs(self.implant)
+
+        # Calculate the max of every current spread map
+        lmax = np.zeros((2, ecs.shape[-1]))
+        if 'INL' in dolayers:
+            lmax[0, :] = ecs[:, :, 0, :].max(axis=(0, 1))
+        if 'NFL' in dolayers:

```

Comment:
Maybe we can start moving towards using "OFL" instead of "NFL"?

`if 'NFL' in dolayers or 'OFL' in dolayers:`

etc.

Preferably remove `NFL` altogether. We don't need to worry about back compatibility for this -- to internal for our current users.